src/components/Table.js

```diff
@@ -0,0 +1,52 @@
+import React, { useState, useEffect, useCallback } from "react"
+
+import { formatAmount, formatDate } from "../utils/formaters"
+
+const Table = ({ deals, selected }) => {
+  const [currentDeals, setCurrentDeals] = useState(deals)
+
+  const handleFilter = useCallback(() => {
+    if (selected) {
+      const filteredDeals = deals.filter((deal) => formatDate(deal.createdAt) === selected)
+      setCurrentDeals(filteredDeals)
+    } else {
+      setCurrentDeals(deals)
+    }
+  }, [selected, deals])
+
+  useEffect(() => {
+    setCurrentDeals(deals)
+  }, [deals])
+
+  useEffect(() => {
+    handleFilter()
+  }, [handleFilter])
+
+  return (
+    <table cellSpacing={0}>
+      <thead>
+        <tr>
+          <th>Title</th>
+          <th>Amount required</th>
+          <th>Created at</th>
+          <th>Updated at</th>
+        </tr>
+      </thead>
+
+      <tbody>
+        {[...currentDeals]
+          .sort((a, b) => a.createdAt < b.createdAt)
```

Comment:
can the sorting be done server side?

Reply:
Yes. I think I could've done it on its own function inside the component before rendering the elements.